emission/tests/modellingTests/TestRunGreedyModel.py

```diff
@@ -0,0 +1,164 @@
+            logging.debug(f"inserting mock Confirmedtrips into database")
+            
+            # generate labels with a known sample weight that we can rely on in the test
+            label_data = {
+                "mode_confirm": ['ebike', 'bike'],
+                "purpose_confirm": ['happy-hour', 'dog-park'],
+                "replaced_mode": ['walk'],
+                "mode_weights": [0.9, 0.1],
+                "purpose_weights": [0.1, 0.9]
+            }
+
+            train = etmm.generate_mock_trips(
+                user_id=user_id,
+                trips=self.total_trips,
+                origin=self.origin,
+                destination=self.destination,
+                label_data=label_data,
+                within_threshold=self.clustered_trips,  
+                threshold=0.004, # ~400m
+                has_label_p=self.has_label_percent
+            )
+
+            ts.bulk_insert(train)
+
+            # confirm data write did not fail
+            test_data = esda.get_entries(key="analysis/confirmed_trip", user_id=user_id, time_query=None)
+            if len(test_data) != self.total_trips:
+                logging.debug(f'test invariant failed after generating test data')
+                self.fail()
+            else:
+                logging.debug(f'found {self.total_trips} trips in database')
+
+    def tearDown(self):
+        """
+        delete entries for user self.user_id in the database, not
+        yet implemented in database operations, so these test entries will
+        have to stick around for now.
+        """
+        pass
+
+    def testBuildGreedyModelFromConfig(self):
+        """
+        greedy model takes config arguments via the constructor for testing
+        purposes but will load from a file in /conf/analysis/ which is tested here
+        """
+
+        eamumt.ModelType.GREEDY_SIMILARITY_BINNING.build()
+        # success if it didn't throw
+
+    def testTrainGreedyModelWithZeroTrips(self):
+        """
+        greedy model takes config arguments via the constructor for testing
+        purposes but will load from a file in /conf/analysis/ which is tested here
+        """
+
+        # making an assumption here...
+        unused_user_id = 'asdjfkl;asdfjkl;asd08234ur13fi4jhf2103mkl'
+
+        # pass along debug model configuration
+        greedy_model_config = {
+            "metric": "od_similarity",
+            "similarity_threshold_meters": 500,
+            "apply_cutoff": False,
+            "incremental_evaluation": False
+        }
+
+        logging.debug(f'~~~~ do nothing ~~~~')
+        eamur.update_trip_model(
+            user_id=unused_user_id,
+            model_type=eamumt.ModelType.GREEDY_SIMILARITY_BINNING,
+            model_storage=eamums.ModelStorage.DOCUMENT_DATABASE,
+            min_trips=self.min_trips,
+            model_config=greedy_model_config
+        )
+
+        # todo: check the pipeline for this user to confirm they don't have a current timestamp
```

Comment:
Can you please fix this todo?

Reply:
updated with the correct pipeline test here (which i just didn't know how to do yet when i wrote this originally, i think)